src/main/java/org/jenkinsci/plugins/github_branch_source/OriginGitHubSCMSource.java

```diff
@@ -88,6 +88,10 @@
         listener.getLogger().format("  %d branches were processed%n", branches);
     }
 
+    @Override protected SCMRevision doRetrieve(SCMHead head, TaskListener listener, GHRepository repo) {
+        return new SCMRevisionImpl(head, "heads/" + head.getName());
```

Comment:
:bug: you need to specify the actual commit hash here. Otherwise what would be the point?